improver/calibration/__init__.py

```diff
@@ -75,30 +84,40 @@ def split_forecasts_and_truth(
             Missing truth or historical forecast in input cubes.
     """
     grouped_cubes = {}
+    land_sea_mask = []
+    static_additional_predictors = []
     for cube in cubes:
-        try:
-            cube_name = get_diagnostic_cube_name_from_probability_name(cube.name())
-        except ValueError:
-            cube_name = cube.name()
-        grouped_cubes.setdefault(cube_name, []).append(cube)
+        if land_sea_mask_name and cube.name() == land_sea_mask_name:
+            land_sea_mask.append(cube)
```

Comment:
I don't think this change can be made, as the input `cubes` is a list of cubes rather than a cubelist, meaning that the `.extract_cube()` method is not available.

Reply:
You're quite right it doesn't work. I feel like there should be a more efficient way of doing this but I can't see what it is and this works . 

It does occur to me that currently if a land_sea_mask_name is defined but there isn't a cube matching that name in the list then there won't be an error which might mean you think your calibrating a cube with a land_sea_mask even though you aren't. It would be worth adding an error raised in this situation.

Reply:
@mspelman07 and I have discussed that, as it currently stands, where a `land_sea_mask_name` is given but there is not a land-sea mask in the `cubes` input an error will not be raised.

We decided that the current (permissive) behaviour is preferable to having an error raised, as it can allow users to feed different configuration of `cubes` input into this function (e.g with/without land-sea mask) without requiring different function calls to be explicitly specified.